src/ProxyManager/ProxyGenerator/NullObject/MethodGenerator/StaticProxyConstructor.php

```diff
@@ -35,6 +35,7 @@ function (ReflectionProperty $publicProperty) : string {
             Properties::fromReflectionClass($originalClass)->getPublicProperties()
         );
 
+        $this->setReturnType($originalClass->getName());
```

Comment:
It should return the proxy class name, not the original class name. Is that feasible, or is that context missing?

Reply:
Well, since the proxy class extends the original class, I think there's no problem with that approach.

Reply:
That would be inaccurate: can we try using the exact type instead, please?